src/ripple/app/paths/cursor/DeliverNodeReverse.cpp

```diff
@@ -37,15 +37,18 @@ namespace path {
 
 // To deliver from an order book, when computing
 TER PathCursor::deliverNodeReverse (
-    AccountID const& uOutAccountID,  // --> Output owner's account.
-    STAmount const& saOutReq,      // --> Funds requested to be
-                                   // delivered for an increment.
-    STAmount& saOutAct) const      // <-- Funds actually delivered for an
-                                   // increment.
+    AccountID const& uOutAccountID, // --> Output owner's account.
+    STAmount const& saOutReq,       // --> Funds requested to be
+                                    // delivered for an increment.
+    STAmount& saOutAct,             // <-- Funds actually delivered for an
+                                    // increment.
+    bool restartDir) const
 {
     TER resultCode   = tesSUCCESS;
 
-    node().directory.restart(multiQuality_);
+    if (restartDir)
+        for (int i = nodeIndex_; i >= 0 && !node (i).isAccount(); --i)
```

Comment:
@rec The reason we go in reverse and stop when we hit an account has to do with the way recursive to `deliverNodeReverse` work. 

When we process an offer node, we also process the run of offer nodes that preceded it (stopping when we hit an account). The directories of those nodes may be effected and need to be reset here. I'll try to come up with a good comment for that explains what's going on here.